src/FluentValidation.DependencyInjectionExtensions/ServiceCollectionExtensions.cs

```diff
@@ -52,13 +52,21 @@ public static IServiceCollection AddValidatorsFromAssemblyContaining<T>(this ISe
 		/// <returns></returns>
 		private static IServiceCollection AddScanResult(this IServiceCollection services, AssemblyScanner.AssemblyScanResult scanResult, ServiceLifetime lifetime)
 		{
+			//Register as interface
 			services.Add(
 				new ServiceDescriptor(
 							serviceType: scanResult.InterfaceType,
 							implementationType: scanResult.ValidatorType,
 							lifetime: lifetime));
 
+			//Register as self

```

Comment:
Added self registration so resolving by TValidator is also possible

Reply:
This introduces an inconsistency with the registration as part of the MVC `AddFluentValidation` call in `FluentValidation.AspNetCore`. The two need to be consistent, so a few options:

- `AddFluentValidation` registrations should pass through these new methods (means FV.AspNetCore has to take a dependency on FV.DI)
- The same change needs to be made manually to FV.AspNetCore
- Or this needs to be removed from here again (for now)

Reply:
I went with option 1 because you already had the dependency there :)

Reply:
Sounds good, thanks!